freedomofpress/securedrop-workstation
# Implements Debian9-based custom templates for all AppVMs

Converts all the AppVMs to reference a custom TemplateVM caused `sd-workstation-template`, which is cloned from the system `debian-9` TemplateVM. Summary of changes:

* All AppVMs are now based on `<app_vm_name>-template`, with a single exception: `sd-gpg` has no corresponding `sd-gpg-template`, because its customizations are simply a keyring that lives in $HOME, and therefore would not be inherited from the TemplateVM.
* Creates `sd-workstation-template` as a base TemplateVM for use across the multiple AppVM configurations. Does _not_ include much of the hardening documented in #156, we can follow up after these changes are merged.
* Script paths have changed throughout to convert `/usr/local/` to `/usr/` equivalents. Qubes handles `/usr/local/` directories specially, and allows them to be unique per VM.
* Similarly, use of `/home/user/` paths is now discouraged, since AppVMs don't inherit those from TemplateVMs. Wherever possible, appropriate system paths are now used. 
* VM config tests expanded somewhat to confirm new template settings

### Testing
The test plan remains the same as usual:

1. Run `make all` in `dom0` and confirm no errors.
2. Run `make test` in `dom0` and confirm no errors.
3. Run **Qubes Manager** and visually confirm all the `sd-` VMs are running Debian 9 (or Whonix 14, which is itself based on Debian 9). 
4. Confirm you can download and view a test submission. A plaintext submission should be displayed in `gedit` _in a DisposableVM_. 

As usual, 4 is the real clincher here. Without any out-of-band modifications, you should be able to view a plaintext submission. Have not tried other file formats yet; we should write up a test plan to account for those.

Closes #131. Related to #156 and #158.
